### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

File: forms/models.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+        return data_fields
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+
+        # If you need to show results only on landing page,
+        # you may need check request.method
+
+        results = dict()
```
Comment (zerolab) [inline forms/models.py:163]: considering the form page has a `show_results` boolean, this whole bit should be conditional.

I would do:

```python
if not self.show_results:
    # return early, without further processing
    return context

results = dict()
# ...
```

File: forms/templates/forms/form_page_landing.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% load wagtailcore_tags %}
+<html>
+    <head>
+        <title>{{ page.title }}</title>
+    </head>
+    <body>
+        <h1>{{ page.title }}</h1>
+
+        <h2>Results</h2>
```
Comment (zerolab) [inline forms/templates/forms/form_page_landing.html:9]: suggestion: wrap this in a `{% if results %}`

File: forms/tests.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (zerolab) [inline forms/tests.py:3]: would be good to have a test case for this:

1. for the logged in/active, or allow anonymous submissiont
2. show/don't show results
3. multi-step would be a nice addition

File: profiles/migrations/0002_auto_20211206_1945.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Generated by Django 3.1.13 on 2021-12-06 19:45
```
Comment (zerolab) [inline profiles/migrations/0002_auto_20211206_1945.py:1]: I can't see where this migration is coming from in the changes from this PR. 
Was this a missing migration from #11 or some other work in progress? Would be good to address it separately.

Additionally, I would recommend using named migrations. ie rename `profiles/migrations/0002_auto_20211206_1945.py` to `profiles/migrations/0002_terms_and_conditions_changes.py` or something more descriptive. It makes looking at the migrations more intuitive

**Review by zerolab — CHANGES_REQUESTED** [review]
Lots of good things here @codiebeulaine 

Left a number of comments and suggestions

File: forms/templates/forms/form_page.html [hunk 5]
```diff
@@ -12,7 +12,7 @@ <h1>{{ page.title }}</h1>
                 <form action="{% pageurl page %}{% if page.multi_step %}?p={{ fields_step.number|add:"1" }}{% endif %}" method="POST">
                     {% csrf_token %}
                     {{ form.as_p }}
-                    <input type="submit">
+                    <button type="submit">{% trans "Submit" %}</button>
```
Comment (zerolab) [inline forms/templates/forms/form_page.html:15]: this needs `i18n` added to the `{% load ` line

Comment (codiebeulaine) [reply forms/templates/forms/form_page.html:15]: 👍

File: profiles/migrations/0002_auto_20211206_1945.py (hunk 4 above)
Comment (codiebeulaine) [reply profiles/migrations/0002_auto_20211206_1945.py:1]: 👍

File: forms/models.py (hunk 1 above)
Comment (codiebeulaine) [reply forms/models.py:165]: 👍

File: forms/templates/forms/form_page_landing.html [hunk 6]
```diff
@@ -5,11 +5,13 @@
...
     <body>
         <h1>{{ page.title }}</h1>
-
-        <h2>Results</h2>
-        {% for question, answers in results.items %}
-            <h3>{{ question }}</h3>
-            {% for answer, count in answers.items %}
-                <div>{{ answer }}: {{ count }}</div>
+        
+        {% if Results %}
```
Comment (zerolab) [inline forms/templates/forms/form_page_landing.html:9]:
```suggestion
        {% if results %}
```

**Comment by KaitCrawford** [comment]
@zerolab :wave: I hope you had a relaxed festive season.
When you have a chance could you please review the changes that have been made to this PR and indicate if you are happy with them. I've added the requested tests in another PR (based off this branch) so that this one doesn't get out of hand since I'm having some trouble with them. https://github.com/girleffect/ge-web/pull/32

**Review by zerolab — APPROVED** [review]
Hey @KaitCrawford @codiebeulain, happy New Year!

This LGTM now. 

Was initially not so sure about the output on the homepage, but looking at the template that looks like it is WIP, so no more worries.

**Comment by KaitCrawford** [comment]
@zerolab thanks! Yeah, the templates are WIP. We're meeting with Mitso tomorrow to check in and I'll merge this PR after I've confirmed that she's happy with that process

**Pull request merged**